geopandas/tests/test_sindex.py

```diff
@@ -371,11 +379,30 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
             ),  # contains but does not contains_properly
         ),
     )
-    def test_query(self, predicate, test_geom, expected):
+    def test_query(self, predicate, test_geom, expected, output_format):
         """Tests the `query` method with valid inputs and valid predicates."""
         res = self.df.sindex.query(test_geom, predicate=predicate)
         assert_array_equal(res, expected)
 
+        if output_format != "indices":
+            dense = np.zeros(len(self.df), dtype=bool)
+            dense[expected] = True
+
+        if output_format == "dense":
+            res = self.df.sindex.query(
+                test_geom, predicate=predicate, output_format=output_format
+            )
+            assert_array_equal(res, dense)
+
+        elif output_format == "sparse":
+            if not HAS_SCIPY:
+                pytest.skip("scipy is required for sparse output")
+            else:
+                res = self.df.sindex.query(
+                    test_geom, predicate=predicate, output_format=output_format
+                )
+                assert_array_equal(res.todense(), dense)
```

Comment:
```suggestion

          res = self.df.sindex.query(
              test_geom, predicate=predicate, output_format=output_format
          )
          if output_format =="sparse":
              res = res.todense()
          assert_array_equal(res, dense)
```
with the sparse skipping handled in the mark this can all be collapsed down I think.